scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -774,10 +774,12 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
 
   /**
    * Assign timestamps to values.
+   * With a optional skew
    * @group window
    */
-  def timestampBy(f: T => Instant): SCollection[T] =
-    this.parDo(FunctionsWithWindowedValue.timestampFn(f))
+  def timestampBy(f: T => Instant, skew: Duration = Duration.ZERO): SCollection[T] =
```

Comment:
I'd name the 2nd argument `allowedTimestampSkew` to match DF API. Plus it's clearer when used as a named arg.

Would be nice to remove `FunctionsWithWindowedValue.timestampFn` which I believe is unused now.